### PR TITLE
Rename TimeZoneTransformer into TimezoneTransformer

### PR DESCRIPTION
**Pull request by lepiaf**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #11028
| License       | MIT
| Doc PR        | no
#

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@lepiaf thanks for contributing a fix for this super old issue report.

To give some context:

1) Symfony uses `Timezone` sometimes and `TimeZone` others.
2) PHP uses "TimeZone" in its classes: http://php.net/manual/en/class.intltimezone.php
3) In UK English, "timezone" doesn't exist (it's "time zone") but in US English, it's more common to use "timezone", so the camelized version should be "Timezone".

Symfony always favors US English, so this change looks correct to me!

File: UPGRADE-3.3.md [hunk 1]
```diff
@@ -177,6 +177,11 @@ HttpKernel
    which will tell the Kernel to use the response code set on the event's
    response object.
 
+Intl
+----
+
+ * Rename `TimeZoneTransformer` class into `TimezoneTransformer`
```
Comment (nicolas-grekas) [inline UPGRADE-3.3.md]: since the class is internal (since 2.8), I'd suggest removing these, and apply this PR on 3.2

Comment (fabpot) [inline UPGRADE-3.3.md]: LGTM

File: UPGRADE-3.2.md [hunk 2]
```diff
@@ -163,6 +163,11 @@ HttpKernel
    Surrogate-Capability: symfony="ESI/1.0"
    ```
 
+Intl
+----
+
+ * Rename `TimeZoneTransformer` class into `TimezoneTransformer`
```
Comment (nicolas-grekas) [inline UPGRADE-3.2.md]: internal class, not needed

File: src/Symfony/Component/Intl/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+CHANGELOG
+=========
+
+3.2.5
+-----
+
+ * Rename `TimeZoneTransformer` class into `TimezoneTransformer`
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/CHANGELOG.md]: internal class, not needed

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Review by xabbuh — APPROVED** [review]
:+1:

**Comment by Tobion** [comment]
👍

**Comment by fabpot** [comment]
Good catch, thanks @lepiaf.

**Pull request closed without merging**